File: src/main/python/visualphysics/layout/GridTest.py
#!/usr/bin/env python3.3

import unittest
import Grid

class GridTest(unittest.TestCase):

    def setUp(self):
        self.p = Grid.Grid(1, 2)

    def test_x(self):
        self.assertEqual(self.p.x, 1)

    def test_y(self):
        self.assertEqual(self.p.y, 2)

    def test_simple_print(self):
        sp_string = "1 2"
        sp = self.p.simple_print()
        self.assertEqual(sp_string, sp)

    def test_pretty_print(self):
        pp_string = "The grid point is (1, 2)."
        pp = self.p.pretty_print()
        self.assertEqual(pp_string, pp)

if __name__ == '__main__':
    unittest.main()

File: Archive/test/java/org/visualphysics/layout/GridTest.java
/* Copyright 2012 by Douglas Sweetser, sweetser@alum.mit.edu
 * Licensed under the Apache License, Version 2.0
 */

package org.visualphysics.layout;

import org.junit.Test;
import static org.junit.Assert.assertEquals;
import org.junit.Before;

import org.visualphysics.layout.Grid;

/** Class GridTest
 *  junit tests
 *  @author sweetser@alum.mit.edu
 */
public class GridTest {

    Grid g;

    @Before
    public void setUp(){
        this.g = new Grid(1, 2);
    }

    @Test
    public void test_x() {
        assertEquals(this.g.x, 1);
    }

    @Test
    public void test_y() {
        assertEquals(this.g.y, 2);
    }

    @Test
    public void test_simple_print() {
        String sp_string = "1 2";
        String sp = this.g.simple_print();
        assertEquals(sp_string, sp);
    }

    @Test
    public void test_pretty_print() {
        String pp_string = "The grid point is (1, 2).";
        String pp = this.g.pretty_print();
        assertEquals(pp_string, pp);
    }
}
